internal/cmd/operator-sdk/alpha/config3alphato3/convert_config_3-alpha_to_3.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config3alphato3
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"golang.org/x/mod/modfile"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	v3alpha   = "3-alpha"
+	versionRe = regexp.MustCompile(`version:[ ]*(?:")?3-alpha(?:")?`)
+)
+
+func convertConfig3AlphaTo3(cfgBytes []byte) (_ []byte, err error) {
+	cfgObj := make(map[string]interface{}, 5)
+	if err := yaml.Unmarshal(cfgBytes, &cfgObj); err != nil {
+		return nil, err
+	}
+
+	if cfgObj["version"].(string) != v3alpha {
```

Comment:
Very old versions ("1") didn't have this field, you may want to check if it is unset or different to `v3alpha`.

Reply:
I mean that he probably wants to check if this field is present:
```suggestion
	if version, hasVersion := cfgObj["version"]; !hasVersion || version.(string) != v3alpha {
```